database/database.go

```diff
@@ -161,13 +162,14 @@ func (db DatastoreBacked) StatesForDomains(domains []string) (states []DomainSta
 
 	client, datastoreErr := db.backend.NewClient(c, db.projectID)
 	if datastoreErr != nil {
-		return states, datastoreErr
+		return nil, datastoreErr
 	}
 
 	getDomainStates := func(keys []*datastore.Key) ([]DomainState, error) {
 		domainStates := make([]DomainState, len(keys))
-		if err := client.GetMulti(c, keys, domainStates); err != nil {
-			return nil, err
+		getErr := client.GetMulti(c, keys, domainStates)
```

Comment:
The "gotX" naming convention is mostly for tests, I think err is probably better here

Reply:
(and also what you had before of setting the err inside the if statement)

Reply:
Done